src/services/web3/transactions/transaction.builder.ts

```diff
@@ -1,11 +1,17 @@
 import { JsonRpcSigner } from '@ethersproject/providers';
-import { ContractConcern } from './concerns/contract.concern';
 import { RawConcern } from './concerns/raw.concern';
+import {
+  getContractConcern,
+  ContractConcernType,
+} from '@/dependencies/contract.concern';
 
 export class TransactionBuilder {
+  contract!: ContractConcernType;
   constructor(
     public readonly signer: JsonRpcSigner,
-    public readonly contract = new ContractConcern(signer),
     public readonly raw = new RawConcern(signer)
-  ) {}
+  ) {
+    const ContractConcernConstructor = getContractConcern();
+    this.contract = new ContractConcernConstructor(this.signer);
+  }
 }
```

Comment:
Could this not still be done as a constructor arg? e.g.
```
const ContractConcernConstructor = getContractConcern();

export class TransactionBuilder {
  constructor(
    public readonly signer: JsonRpcSigner,
    public readonly contract = new ContractConcernConstructor(signer),
    public readonly raw = new RawConcern(signer)
  ) {}
}

Reply:
The problem is that we cannot call `getContractConcern` outside the constructor because it would be called before we set the mock from the tests. 

Another option was trying this but TS doesn't let you do that. 

<img width="828" alt="Screenshot 2023-05-24 at 12 27 39" src="https://github.com/balancer/frontend-v2/assets/1316240/7b44e7a1-d974-42bd-98f6-db72b9c2e6b1">

Reply:
Ok no worries.